src/Data/String/Unsafe.ss

```diff
@@ -0,0 +1,28 @@
+;; -*- mode: scheme -*-
+
+(library (Data.String.Unsafe foreign)
+  (export char charAt)
+  (import
+    (only (rnrs base) = define if lambda string-length string-ref)
+    (only (rnrs exceptions) raise-continuable with-exception-handler)
+    (only (rnrs conditions) make-message-condition))
+
+  (define charAt
+    (lambda (i)
+      (lambda (s)
+        (with-exception-handler
+          (lambda (e)
+            (raise-continuable
+              (make-message-condition
+                "Data.String.Unsafe.charAt: Invalid index.")))
+          (lambda ()
+            (string-ref s i))))))
+
+    (define char
+      (lambda (s)
+        (if (= (string-length s) 1)
+          (string-ref s 0)
+          (raise-continuable
+            (make-message-condition
+              "Data.String.Unsafe.char: Expected string of length 1.")))))
```

Comment:
Can't we just call `string-ref` here and not do the bounds check?

Reply:
We would then, at least, need to catch it anyways to format the error. But that aside, the main reason is that the PS function is explicitly like this: a string of length 1, no more no less, else I throw. This here is direct mirroring of the JS implementation.

Reply:
I think the only reason the JS version of both of these functions throw an exception "manually"  is beause `String.prototype.charAt` itself won't throw an exception:

```
> "asdfa".charAt(100)
''
> "".charAt(0)
''
```

The point of the `Unsafe` module is to be unsafe, so I think we should do the bare minimum to return the correct value and not do any special handling. IMO we should remove all catching from this module _and_ also accept that `char` will return a value even if the string has a length of more than 1. `string-ref` will itself throw an exception if it can't return a `Char` at the given index. I would not care about the error message string.